cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeleteAndCreateSchedulesStage.java

```diff
@@ -37,25 +42,26 @@
 public class DeleteAndCreateSchedulesStage extends AbstractStage<ApplicationWithPrograms> {
 
   private final Scheduler programScheduler;
+  private final ProfileStore profileStore;
 
-  public DeleteAndCreateSchedulesStage(Scheduler programScheduler) {
+  public DeleteAndCreateSchedulesStage(Scheduler programScheduler, ProfileStore profileStore) {
```

Comment:
I think we should do the profile checking in a separate stage, rather than tie it to the schedules stage. Profiles should be looked up just once during the deployment process, otherwise you can have very weird races where it's disabled in one stage but not in the other.

Reply:
We actually do not need this check here, since the CoreSchedulerService will get called when we add schedules when we deploy the app, and there the status will be called once and in the same transaction with adding the profiles.